### PR TITLE
Add Slack notification action to workflows

### PR DESCRIPTION
**Pull request by mtyazici**
This PR adds Slack notification action to all the workflows.

If a workflow finishes unsuccessfully, it will create a web-hook for the Slack channel `github-actions-log`.

### CODE REVIEW

File: .github/workflows/hz-enterprise-op.-rhel-release.yaml
Comment (leszko) [inline .github/workflows/hz-enterprise-op.-rhel-release.yaml:385]: Do we want to get the notification always or only if the pipeline fails? WDYT? @hasancelik any opinion on that?

File: .github/workflows/hz-enterprise-op.-rhel-release.yaml
Comment (leszko) [inline .github/workflows/hz-enterprise-op.-rhel-release.yaml:394]: In terms of security, do we have any policy what GH Actions we use and what credentials we can pass to them? I remember that it was a concern of @jerrinot some time ago.

CC: @kwart

File: .github/workflows/hz-enterprise-op.-rhel-release.yaml
Comment (mtyazici) [reply .github/workflows/hz-enterprise-op.-rhel-release.yaml:385]: Actually this statement is only there to run the Slack job if the first one doesn't succeed. In the Slack job there is only one step and it has the following condition `if: needs.build_publish.result != 'success'`. So, we will only get notifications when the pipeline fails. I tried to write the condition at job scope, but it didn't work so I did this workaround.

File: .github/workflows/hz-enterprise-op.-rhel-release.yaml
Comment (mtyazici) [reply .github/workflows/hz-enterprise-op.-rhel-release.yaml:394]: `GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}` This token has scope only for this repository and it is created default by Github.

Comment (kwart) [reply .github/workflows/hz-enterprise-op.-rhel-release.yaml:394]: AFAIK we don't have such a policy yet.

What we should probably do is to create a list of trusted organizations (i.e. github accounts). Actions from these trusted accounts would be allowed to work with sensitive information.
If we want to use an action from a user/organization which is not listed and the action would use a sensitive information, then we should fork the action code, review it and reference the action from our own repository.

Comment (leszko) [reply .github/workflows/hz-enterprise-op.-rhel-release.yaml:394]: @mtyazici Could you please do the research? I mean to do the following:

1. Check what possible GH Actions we can use for the Slack Notifications.
2. Check their popularity / usage (how many stars, what other tools use them)
3. Ping Josef on slack and discuss if we should just use the given action or should we fork the repo

Comment (mtyazici) [reply .github/workflows/hz-enterprise-op.-rhel-release.yaml:394]: Sure, I will

**Review by leszko — APPROVED** [review]
LGTM 👍 

Please wait for the review from @kwart before merging.

**Review by kwart — APPROVED** [review]
LGTM :+1:
The specific commit is used for the GH action. The commit was reviewed by Muhammet.

**Pull request merged**
